keps/sig-scheduling/20190311-resource_bin_packing_priority_function.md

```diff
@@ -0,0 +1,175 @@
+---
+title: Resource Bin Packing Priority Function
+  - "@sudeshsh"
+owning-sig: sig-scheduling
+participating-sigs:
+  - sig-scheduling
+reviewers:
+  - "@k82cn"
+  - "@Huang-Wei"
+approvers:
+  - "@k82cn"
+creation-date: 2019-03-11
+last-updated: 2019-03-11
+status: provisional
+---
+
+# Resource Bin Packing Priority Function
+
+## Table of Contents
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Story 1](#story-1)
+- [Design Details](#design-details)
+  - [Examples](#examples)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+
+## Summary
+
+Resource Bin Packing Priority Function allows users to use the best fit polices during scheduling. It allows users to apply [bin packing](https://en.wikipedia.org/wiki/Bin_packing_problem) on core resources like CPU, Memory as well as extended resources like accelerators.
+
+## Motivation
+
+While running Machine Learning workloads on kubernetes which use accelerator devices the default scheduler spreads pods across nodes resulting in fragmentation of extended resources. This fragmentation prevents scheduling pods with larger device resource requirements, and they remain in pending state.
+
+### Goals
+
+- Schedule Jobs using BestFit Policy using Resource Bin Packing Priority Function
```

Comment:
in default scheduler, we only schedule pod :)

Reply:
👍